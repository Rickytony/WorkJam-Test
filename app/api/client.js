import axios from "axios";

const client = axios.create({
  baseURL: "https://hacker-news.firebaseio.com/v0"
});

export { client };
