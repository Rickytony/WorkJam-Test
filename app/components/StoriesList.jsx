import React from "react";
import { getStory, getTopStories } from "../api/client.js";
import { Story } from "./Story.jsx";

class StoriesList extends React.Component {
  constructor() {
    super();

    this.state = {
      stories: []
    };
    this.getStories = this.getStories.bind(this);
  }

  componentDidMount() {
    return getTopStories()
      .then(response => {
        this.getStories(response.data.slice(0, 10));
      })
      .catch(error => {
        throw new Error(error);
      });
  }

  getStories(storyIds) {
    storyIds.forEach(story => {
      return getStory(story).then(response => {
        this.setState({
          stories: [...this.state.stories, response.data]
        });
      });
    });
  }

  render() {
    if (this.state.stories.length < 1) {
      return (
        <div className="container">
          <h2 className="col">Stories are loading...</h2>
        </div>
      );
    }

    return (
      <div className="container mt-4">
        {this.state.stories.map(story => (
          <Story data={story} key={story.id} />
        ))}
      </div>
    );
  }
}

export { StoriesList };
