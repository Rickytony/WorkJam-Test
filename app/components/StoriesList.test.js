import React from "react";
import { shallow } from "enzyme";
import * as client from "../api/client";
import { StoriesList } from "./StoriesList";

jest.mock("../api/client");

const story = { title: "story title", by: "author", id: 1 };
const commentIds = { data: [1] };
client.getTopStories = jest.fn().mockResolvedValue(commentIds);
client.getStory = jest.fn().mockResolvedValue({ data: story });

describe("StoriesList", () => {
  it("renders correctly if there are no stories", () => {
    const wrapper = shallow(<StoriesList />);
    expect(wrapper.find("h2").text()).toEqual("Stories are loading...");
  });

  it("renders correctly", () => {
    const wrapper = shallow(<StoriesList />);
    wrapper.setState({ stories: [story] });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
