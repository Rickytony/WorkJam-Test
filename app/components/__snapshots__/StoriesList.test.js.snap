// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoriesList renders correctly 1`] = `ShallowWrapper {}`;
